polyfill/lib/calendar.mjs

```diff
@@ -1397,9 +1394,13 @@ const makeHelperGregorian = (id, originalEras) => {
     },
     /** Fill in missing parts of the (year, era, eraYear) tuple */
     completeEraYear(calendarDate) {
-      const checkField = (name, value) => {
+      const checkField = (name, value, aliases) => {
         const currentValue = calendarDate[name];
-        if (currentValue != null && currentValue != value) {
+        if (
+          currentValue != null &&
+          currentValue != value &&
+          !ES.Call(ArrayPrototypeIncludes, aliases || [], [currentValue])
+        ) {
```

Comment:
I think this change (and the corresponding adaptation to return `eraAliases` below) could use some more attention in tests. I believe it's correct and necessary, but there aren't any tests that fail if I revert it. I propose that we still go ahead and merge this but if you have an idea for a test case, please add it to test262 in a follow up.

Reply:
Hmm, let's see... without this use of `aliases`, a call like `Temporal.PlainDate.from({ calendar: "gregory", era: "ce", eraYear: 2024, year: 2000, month: 1, day: 1 })` (having year along with era and eraYear) fails with RangeError "_Input era ce doesn't match calculated value gregory_". So there's your test case input, although I'm not sure where to put it in test262 so can I ask you to do that and I'll review?

Also, I feel like the polyfill has terrible choices for the primary names of Gregorian calendar eras—the above should probably read like "_…doesn't match calculated value ce_" or "_…doesn't match calculated value ad_", and the message for `Temporal.PlainDate.from({ calendar: "gregory", era: "ce", eraYear: 0, year: 0, month: 1, day: 1 })` should probably be "_Input era ce doesn't match calculated value bce_" or "_Input era ce doesn't match calculated value bc_" rather than "_Input era ce doesn't match calculated value gregory-inverse_" because what the heck is "gregory-inverse" to the median developer?

Reply:
Thanks for the test case. This is done in https://github.com/tc39/test262/pull/4232. I checked that it fails when you revert the aliases fix.

Re. the era names, our hands are tied, as I'm trying to follow the current state of the [Intl Era and Month Codes proposal](https://tc39.es/proposal-intl-era-monthcode/#table-eras). Justin has a [ticket open](https://github.com/tc39/proposal-intl-era-monthcode/issues/12) for improving the canonical names. I guess in the polyfill we could prefer printing an alias in the error messages if there is one.

Reply:
Thanks; test262 PR approved.

And I just now left a comment on the era/month code issue; we're right to keep the polyfill aligned with it but hopefully that part will improve.

Reply:
Maybe we could show both the month code proposal names and the more recognizable alias(es) in the error message?

Reply:
Tentatively, that's what we're doing now since #2944. If the era/monthcode proposal changes, we can change it to do something else.